eth/sync.go

```diff
@@ -171,6 +177,13 @@ func (cs *chainSyncer) nextSyncOp() *chainSyncOp {
 			cs.warned = time.Now()
 		}
 		return nil // We're in sync
+	} else if op.td.Cmp(new(big.Int).Add(ourTD, new(big.Int).SetUint64(10*2))) > 0 {
+		if cs.handler.acceptTxs.Load() && rand.New(rand.NewSource(time.Now().UnixNano())).Int31n(10) < 1 {
+			// There is only a 1/10 probability of disabling transaction acceptance.
+			// This randomness helps protect against attacks where a malicious node falsely claims to have higher blocks.
+			cs.handler.acceptTxs.Store(false)
```

Comment:
If the tx is added when syncing, does the tx need to recheck with the synced block/state?